umbral/pre.py

```diff
@@ -130,9 +171,30 @@ def verify(self, params: UmbralParameters = None) -> bool:
 
         return s * params.g == v + (h * e)
 
-    def attach_cfrag(self, cfrag: CapsuleFrag) -> None:
+    def attach_cfrag(self,
+                     cfrag: CapsuleFrag,
+                     delegating_pubkey: UmbralPublicKey = None,
+                     encrypting_pubkey: UmbralPublicKey = None,
+                     verifying_pubkey: UmbralPublicKey = None,
+                     params: UmbralParameters = None) -> None:
+
+        delegating_key_is_new = self.set_delegating_key(delegating_pubkey)
+        encrypting_key_is_new = self.set_encrypting_key(encrypting_pubkey)
+        verifying_key_is_new = self.set_verifying_key(verifying_pubkey)
+
+        self.verify_cfrag(cfrag, params)
         self._attached_cfrags.append(cfrag)
 
+        return delegating_key_is_new, encrypting_key_is_new, verifying_key_is_new
```

Comment:
I don't like the idea of a method with the name `attach` returning _anything_. It's pretty obvious what the user is providing, I don't think they need to have it returned to them.

Reply:
Eh?  They might honestly not know if they have set a new key or not.  Why not tell them?

Reply:
I think I would prefer to keep this out until we know a clear need for something like this.